core/src/test/java/com/sequenceiq/cloudbreak/service/StackCommonServiceTest.java

```diff
@@ -193,51 +198,44 @@ public void testDeleteInstancesFromDataHub() {
-        StackScaleV4Request updateRequest = new StackScaleV4Request();
-        updateRequest.setStackNetworkScaleV4Request(networkScaleV4Request);
 
-        Assertions.assertThrows(BadRequestException.class, () -> underTest.putScalingInWorkspace(STACK_NAME, WORKSPACE_ID, updateRequest));
+        Assertions.assertThrows(BadRequestException.class, () -> underTest.validateNetworkScaleRequest(stack, networkScaleV4Request));
 
         Mockito.verify(multiAzValidator, times(1)).supportedVariant(variant);
         Mockito.verify(multiAzValidator, times(0)).collectSubnetIds(any());
     }
 
     @Test
-    public void testPutScalingInWorkspaceWhenThereIsPreferredAzAndStackProvisionedToASingleSubnet() {
+    public void testValidateNetworkScaleRequestWhenThereIsPreferredAzAndStackProvisionedToASingleSubnet() {
         Stack stack = new Stack();
         String variant = AwsConstants.AwsVariant.AWS_NATIVE_VARIANT.name();
         stack.setPlatformVariant(variant);
-        when(stackService.getByNameOrCrnInWorkspace(STACK_NAME, WORKSPACE_ID)).thenReturn(stack);
         when(multiAzValidator.supportedVariant(variant)).thenReturn(Boolean.TRUE);
         when(multiAzValidator.collectSubnetIds(any())).thenReturn(Set.of(SUBNET_ID));
         NetworkScaleV4Request networkScaleV4Request = new NetworkScaleV4Request();
         networkScaleV4Request.setPreferredSubnetIds(List.of(SUBNET_ID));
-        StackScaleV4Request updateRequest = new StackScaleV4Request();
-        updateRequest.setStackNetworkScaleV4Request(networkScaleV4Request);
 
-        Assertions.assertThrows(BadRequestException.class, () -> underTest.putScalingInWorkspace(STACK_NAME, WORKSPACE_ID, updateRequest));
+        Assertions.assertThrows(BadRequestException.class, () -> underTest.validateNetworkScaleRequest(stack, networkScaleV4Request));
 
         Mockito.verify(multiAzValidator, times(1)).supportedVariant(variant);
         Mockito.verify(multiAzValidator, times(1)).collectSubnetIds(any());
     }
 
     @Test
-    public void testPutScalingInWorkspaceWhenThereIsPreferredAzAndStackProvisionedInMultipleSubnetButScalingIsNotSupportedOnPlatform() {
+    public void testPutScalingInWorkspaceWhenThereIsPreferredAzAndStackProvisionedInMultipleSubnetButScalingIsNotSupportedOnPlatform()
+            throws TransactionService.TransactionExecutionException {
         Stack stack = new Stack();
         stack.setCloudPlatform(AwsConstants.AWS_PLATFORM.value());
         String variant = AwsConstants.AwsVariant.AWS_NATIVE_VARIANT.name();
         stack.setPlatformVariant(variant);
-        when(stackService.getByNameOrCrnInWorkspace(STACK_NAME, WORKSPACE_ID)).thenReturn(stack);
-        when(multiAzValidator.supportedVariant(variant)).thenReturn(Boolean.TRUE);
-        when(multiAzValidator.collectSubnetIds(any())).thenReturn(Set.of(SUBNET_ID, "anotherSubnetId"));
+        when(transactionService.required(any(Supplier.class))).thenReturn(stack);
```

Comment:
Not a big deal but I suggest stubbing `transactionService.required()` so that it actually executes the provided `Supplier` lambda. That way `stackService.getByNameOrCrnInWorkspace()` would have to be stubbed again, and of course the same goes for `multiAzValidator.*`.

Reply:
is it matter? We want the stack object only.

Reply:
yes, it is the only advantage of the `@Spy` but I have changed the tests to call directly the tested method